### DISCUSSION TITLE
v2/hsa/targets/report unrecognizd metric

### DISCUSSION THREAD

**Discussion by buraksaler** [amzn/ads-advanced-tools-docs#252] [Q&A]
Hi, I'm trying to use v2 target reporting and the Target metrics I got from the [docs](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v2/metrics) are unrecognized. Even the metrics in the sample request. How can I solve it? thanks in advance

the error is: '{"code":"422","details":"Unrecognized metric:  impressions\\n at [Source:","requestId":"1GR1VMRE5ACQW42MJQZB"}'

**Comment by elizafarley** [comment]
@buraksaler can you share the request body resulting in an error? 

This request is working for me:

```
curl --location 'https://advertising-api.amazon.com/v2/hsa/campaigns/report' \
--header 'Amazon-Advertising-API-ClientId: amzn1.application-oa2-client.xxxxxxx' \
--header 'Amazon-Advertising-API-Scope: xxxxx' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer Atza|xxxxxxx' \
--data '{
    "reportDate":"20240410",
    "metrics":"impressions,clicks,cost,attributedSales14d,attributedConversions14d"
}'
```

**Reply by buraksaler** [reply]
@elizafarley the example that you give is campaign report. Campaign reports are working for me. Target report type is giving me an error. My request body is like this:

{
'metrics': 'targetId, impressions,clicks,cost', 
'creativeType': 'all',
 'reportDate': '20240417'}

**Reply by elizafarley** [reply]
I think the space before `impressions` is causing the issue here.

Try:

```
{
  'metrics': 'targetId,impressions,clicks,cost',
  'creativeType': 'all',
  'reportDate': '20240417'
}
```

**Reply by buraksaler** [reply]
@elizafarley thank you, I managed creating a report. But for some accounts, although there are data on console, I get an empty array after decoding the response. I use v2 targets for the accounts that I get empty array with v3 Sponsored Brands Targeting. 
I also use v2 campaigns instead of v3 sbCampaigns and there is no problem. 
What could be the reason for targeting?

**Reply by elizafarley** [reply]
V3 reporting for Sponsored Brands is currently available in preview. This is because currently data related to Sponsored Brands campaigns with flag isMultiAdGroupsEnabled=False won’t be available. Campaigns with this flag are legacy SB campaigns built using an older version of the API. We are working on supporting these older campaigns in V3 reporting, but until then, you'll need to use V2 reporting to pull data for older SB campaigns.

**Reply by buraksaler** [reply]
Yes, I am using v2 campaigns and there is no problem. Problem is when I try to get v2 targets report, I get an empty array. But there are data on the console.

**Reply by elizafarley** [reply]
In the console there is only one target report that contains both keyword and product target data. In the v2 reporting API, there are two separate reports, one for keywords and one for targets. Are you requesting both of these reports via API and comparing the aggregate to the console?

See [here](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v2/advertising-console) for more info.
